Composed/Core/DataSources/SingleElementDataSource.swift

```diff
@@ -0,0 +1,101 @@
+/**
+ A `DataSource` that contains a single element. This DataSource will always
+ return `1` for `numberOfSections`.
+ */
+open class SingleElementDataSource<Element>: DataSource {
+
+    /// A delegate that will be notified of updates to the data source
+    public weak var updateDelegate: DataSourceUpdateDelegate?
+    
+    /// The element being stored by the data source
+    public private(set) var element: Element
+    
+    /// `true` if `element` is an `Optional.none`
+    public var elementIsNil: Bool {
```

Comment:
I don't agree with this being in the lib. Its an easy extension to add on later.
I don't see maybe users actually creating a `SingleElementDataSource<Optional<String>>` for example.
And this only makes sense in those context's.